### ISSUE TITLE
cli: list/destroy: EKS region scan fails prematurely w/o showing error detail

### ISSUE THREAD

**Issue by sreis** [opstrace/opstrace#740]
**Describe the bug**

The cluster is created successfully.
```
[2021-05-14T17:19:38Z] 2021-05-14T17:19:38.456Z info: wait for grafana endpoints: all probe URLs returned expected HTTP responses, continue
[2021-05-14T17:19:38Z] 2021-05-14T17:19:38.457Z info: cluster creation finished: upgrade-bk-598-318-a (aws)
[2021-05-14T17:19:38Z] 2021-05-14T17:19:38.457Z info: Log in here: https://upgrade-bk-598-318-a.opstrace.io
[2021-05-14T17:19:38Z] 2021-05-14T17:19:38.457Z debug: task `cluster creation` succeeded (attempt 1)
[2021-05-14T17:19:38Z] 2021-05-14T17:19:38.457Z debug: end of createCluster()
[2021-05-14T17:19:38Z] 2021-05-14T17:19:38.457Z debug: shut down logger, then exit with code 0
```

But the [list command fails to find it](https://buildkite.com/opstrace/upgrade-tests/builds/598#d85b6962-942f-4049-a7da-8ce55c2056b4/1754-5048) when doing the upgrade. And [again](https://buildkite.com/opstrace/upgrade-tests/builds/598#d85b6962-942f-4049-a7da-8ce55c2056b4/5405-5478) when doing the teardown.
```
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.421Z info: starting lookup of matching EKS cluster accross AWS regions
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.423Z debug: eks.listClusters() for region af-south-1
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.430Z debug: eks.listClusters() for region eu-north-1
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.431Z debug: eks.listClusters() for region ap-south-1
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.433Z debug: eks.listClusters() for region eu-west-3
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.434Z debug: eks.listClusters() for region eu-west-2
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.435Z debug: eks.listClusters() for region eu-south-1
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.436Z debug: eks.listClusters() for region eu-west-1
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.438Z debug: eks.listClusters() for region ap-northeast-2
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.439Z debug: eks.listClusters() for region me-south-1
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.440Z debug: eks.listClusters() for region ap-northeast-1
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.441Z debug: eks.listClusters() for region sa-east-1
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.442Z debug: eks.listClusters() for region ca-central-1
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.443Z debug: eks.listClusters() for region ap-east-1
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.444Z debug: eks.listClusters() for region ap-southeast-1
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.445Z debug: eks.listClusters() for region ap-southeast-2
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.446Z debug: eks.listClusters() for region eu-central-1
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.447Z debug: eks.listClusters() for region us-east-1
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.448Z debug: eks.listClusters() for region us-east-2
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.449Z debug: eks.listClusters() for region us-west-1
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.450Z debug: eks.listClusters() for region us-west-2
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.451Z debug: eks.listClusters() for region cn-north-1
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.453Z debug: eks.listClusters() for region cn-northwest-1
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.960Z debug: aws sdk: [AWS eks 200 0.51s 0 retries] listClusters({})
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.961Z debug: eks.listClusters() for region us-west-1 returned 0 EKS clusters
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.972Z debug: aws sdk: [AWS eks 200 0.53s 0 retries] listClusters({})
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.973Z debug: eks.listClusters() for region ca-central-1 returned 0 EKS clusters
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.990Z debug: aws sdk: [AWS eks 200 0.54s 0 retries] listClusters({})
[2021-05-14T17:19:41Z] 2021-05-14T17:19:41.991Z debug: eks.listClusters() for region us-west-2 returned 9 EKS clusters
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.042Z debug: aws sdk: [AWS eks 200 0.594s 0 retries] listClusters({})
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.043Z debug: eks.listClusters() for region us-east-2 returned 1 EKS clusters
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.066Z debug: aws sdk: [AWS eks 200 0.626s 0 retries] listClusters({})
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.067Z debug: eks.listClusters() for region ap-northeast-1 returned 0 EKS clusters
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.080Z debug: aws sdk: [AWS eks 200 0.644s 0 retries] listClusters({})
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.080Z debug: eks.listClusters() for region eu-west-1 returned 0 EKS clusters
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.082Z debug: aws sdk: [AWS eks 200 0.648s 0 retries] listClusters({})
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.082Z debug: eks.listClusters() for region eu-west-2 returned 0 EKS clusters
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.094Z debug: aws sdk: [AWS eks 200 0.661s 0 retries] listClusters({})
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.094Z debug: eks.listClusters() for region eu-west-3 returned 0 EKS clusters
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.096Z debug: aws sdk: [AWS eks 200 0.105s 0 retries] describeCluster({ name: 'tracy' })
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.102Z debug: aws sdk: [AWS eks 200 0.655s 0 retries] listClusters({})
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.102Z debug: eks.listClusters() for region us-east-1 returned 0 EKS clusters
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.128Z debug: aws sdk: [AWS eks 200 0.698s 0 retries] listClusters({})
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.128Z debug: eks.listClusters() for region eu-north-1 returned 0 EKS clusters
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.143Z debug: aws sdk: [AWS eks 200 0.705s 0 retries] listClusters({})
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.143Z debug: eks.listClusters() for region ap-northeast-2 returned 0 EKS clusters
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.176Z debug: aws-sdk-js request failed (attempt 0): UnrecognizedClientException: The security token included in the request is invalid. (retryable, according to sdk: false), retry in 1.22 seconds
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.176Z debug: aws sdk: [AWS eks 403 0.741s 0 retries] listClusters({})
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.176Z info: listing EKS clusters for region eu-south-1 failed with a 403 error (not authorized), skip region
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.210Z debug: aws sdk: [AWS eks 200 0.113s 0 retries] describeCluster({ name: 'upgrade-bk-598-318-a' })
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.262Z debug: aws-sdk-js request failed (attempt 0): UnrecognizedClientException: The security token included in the request is invalid. (retryable, according to sdk: false), retry in 1.75 seconds
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.263Z debug: aws sdk: [AWS eks 403 0.82s 0 retries] listClusters({})
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.263Z info: listing EKS clusters for region ap-east-1 failed with a 403 error (not authorized), skip region
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.302Z debug: aws sdk: [AWS eks 200 0.092s 0 retries] describeCluster({ name: 'schedul-bk-2286-cfe-a' })
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.340Z debug: aws sdk: [AWS eks 200 0.297s 0 retries] describeCluster({ name: 'sreis-test' })
[2021-05-14T17:19:42Z] 2021-05-14T17:19:42.351Z error: No EKS cluster found for Opstrace cluster name 'upgrade-bk-598-318-a. Assume that the Opstrace cluster does not exist (anymore). You can force running the requested operation in a specific AWS region by setting the --region <region> command line parameter.
```

**Comment by jgehrcke** [comment]
Per region:
* one HTTP request to list EKS clusters
* one HTTP request to inspect _each_ EKS cluster

Once a region is completely scanned via above's HTTP requests, there is a debug log message:

https://github.com/opstrace/opstrace/blob/d24672880039d1e7a33348843f5b2716222c3b40/packages/cli/src/list.ts#L219

This is not shown for any region in the log excerpt you have shown.

Instead we see

```
[2021-05-14T17:19:44Z] 2021-05-14T17:19:44.537Z error: No EKS cluster found for Opstrace cluster name 'upgrade-bk-598-318-a. Assume that the Opstrace cluster does not exist (anymore). You can force running the requested operation in a specific AWS region by setting the --region <region> command line parameter.
```

This means that an error is thrown and the detail is swallowed (nothing logged).

I need to go through all error handlers; clearly a bug somewhere.

**Comment by jgehrcke** [comment]
Fixed via #744.
